source/includes/php-frameworks/Restaurant.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Document;
+
+use DateTime;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ODM\MongoDB\Mapping\Annotations as MongoDB;
+use Doctrine\ODM\MongoDB\Mapping\Annotations as ODM;
+use Doctrine\ODM\MongoDB\Types\Type;
+
+#[
+    MongoDB\Document(collection: 'restaurants')
+]
+
+class Restaurant
+{
+    #[ODM\Id]
+    public  $id;
```

Comment:
It should be a `string`, as ODM's `Id` type converts ObjectIds to strings for better handling in web requests. Off the top of my head I don't know how well Doctrine handles uninitialised properties, so if we want to be cautious it should be a nullable string with default value:

```suggestion
    public ?string $id = null;
```

Reply:
> ODM's Id type converts ObjectIds to strings for better handling in web requests

Yes, of course. I want to come out on record here and say that it's been _that_ long since I worked on an app with ODM that I legitimately forgot about that for a hot minute.

Reply:
Changed the field attribute to `STRING` as well

Reply:
Changing the field definition was fine, but changing the annotation to `#[ODM\Field]` for the `id` field resulting in this error: 
`No identifier/primary key specified for Entity...`
Is this expected? Fixed by changing the annotation back to `#[ODM\Id]`

Reply:
`#[ODM\Id]` is correct. Even though ODM converts ObjectIds to string in the model class, it should be tracked as an identifier.